### PR TITLE
Make chunk cache ttl configurable

### PR DESCRIPTION
**Pull request by PVince81**
Continuation of https://github.com/owncloud/core/pull/23486

Please review @icewind1991 @nickvergessen @schiesbn @rullzer 

CC @guruz

### CODE REVIEW

**Comment by PVince81** [comment]
Fixes https://github.com/owncloud/core/issues/23455

**Comment by PVince81** [comment]
Hmm, I should probably rename the value to "chunk_ttl" since the value now only applies to chunks

**Comment by icewind1991** [comment]
> Hmm, I should probably rename the value to "chunk_ttl" since the value now only applies to chunks

Yes that would be a lot better

:+1: otherwise

**Comment by rullzer** [comment]
@PVince81 wouldn't it make more sense to have some more logic for chunks? 

So update the mtime of all chunks (if it is more than X difference?) so... that way as long as the user resumes upload withing a certain ttl no chunks will be removed. Also makes sense from kind of an abstract point that you never want to delete chunks of a file that you are actively uploading...

File: config/config.sample.php [hunk 1]
```diff
@@ -970,6 +970,14 @@
 'cache_path' => '',
 
 /**
+ * TTL of files located in the cache folder before they're removed by
+ * garbage collection (in seconds). Increase this value if users have
+ * issues uploading very large files via the ownCloud Client as upload isn't
+ * completed within one day.
+ */
+'cache_folder_gc_ttl' => 86400, // 60*60*24 = 1 day
```
Comment (nickvergessen) [inline config/config.sample.php]: > I should probably rename the value to "chunk_ttl" since the value now only applies to chunks

Comment (PVince81) [inline config/config.sample.php]: ok, pushed

**Comment by PVince81** [comment]
@rullzer yes, but that is more complex and can be done separately.

Here I was mostly taking over @fkammer's TTL addition.

**Comment by rullzer** [comment]
Ok. fine by me then :+1:

**Comment by guruz** [comment]
I think this value is too low, especially if there is nothing done like @rullzer had recommended
(e.g. taking the newest chunk as lifetime for all chunks).

Especially if a user shuts down his/her machine at night or is travelling or [...] ?

**Comment by PVince81** [comment]
@guruz this was always the default. Won't change it in this PR. The purpose of this PR is making it configurable.

If you want a higher default value, please raise a separate ticket.

**Comment by guruz** [comment]
👍

**Pull request merged**